pymc3/model.py

```diff
@@ -800,7 +804,36 @@ def cont_vars(self):
         """All the continuous variables in the model"""
         return list(typefilter(self.vars, continuous_types))
 
-    def Var(self, name, dist, data=None, total_size=None):
+    def shape_from_dims(self, dims):
+        shape = []
+        if len(set(dims)) != len(dims):
+            raise ValueError('Can not contain the same dimension name twice.')
```

Comment:
Mostly because it is trouble downstream in my experience. There are a few cases where that would be nice (eg distributions over corr matrices), but if you do that, xarray will start doing strange things. For example, what should `trace.mean('non-unique-dim')` do? We could allow this and just let users deal with the consequences. I usually use a copy of the same coordinates with a slightly modified name in those cases, much less trouble. :-)
But to get started, I though better safe than sorry...

Reply:
This is reasonable thing. Multiple coords is still a valid thing (right?).

Reply:
Multiple coords?

Reply:
Non-unique coords,

    dim TrueFalse
    coord [1, 1, 1, 0, 0, 1]

Reply:
I think you can do that, yes. I can't think of an application where I would want to right now however. Do you have something in mind? Shouldn't something like that usually be a data variable instead of a coordinate?

Reply:
Yeah, usually. (There was nothing special, just wanted to make sure that is still possible)